gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -49,22 +51,98 @@ public void write(Bundle bundle, File rootFolder) {
         });
 
         //create policies
+        Map<String, PolicyMetadata> policyMetadataMap = new HashMap<>();
         Map<String, Service> services = bundle.getEntities(Service.class);
-        services.values().parallelStream().forEach(serviceEntity -> writePolicy(bundle, policyFolder, serviceEntity.getParentFolder().getId(), serviceEntity.getName(), serviceEntity.getPolicyXML()));
+        services.values().parallelStream().forEach(serviceEntity -> {
+            final String id = serviceEntity.getId();
+            final String name = serviceEntity.getName();
+            final Folder folder = bundle.getFolderTree().getFolderById(serviceEntity.getParentFolder().getId());
+            final Path policyPath = bundle.getFolderTree().getPath(folder);
+            writePolicy(policyFolder, policyPath, name, serviceEntity.getPolicyXML());
+            final PolicyMetadata policyMetadata = createPolicyMetadata(id, rawBundle, null);
+            final String path = PathUtils.unixPathEndingWithSeparator(policyPath);
+            final String key = path.length() > 1 ? path + name : name;
+            policyMetadataMap.put(key, policyMetadata);
```

Comment:
Can you move this code to another method instead of having it inline? 
Lots of code duplication in this and the below code. Maybe you can merge them?